radiometric_normalization/time_stack.py

```diff
@@ -0,0 +1,214 @@
+        - all the images have the same nodata value
+
+    The output file is only supposed to be used internally so options to change
+    the nodata value and the datatype are not exposed.
+    - The output nodata value is 2 ** 15 - 1
+    - The output image data type is uint16
+    - Since the output file is only meant to be used internally, the
+        geographic information isn't carried over to the output file.
+
+    Input:
+        image_paths (list of str): A list of the input file paths
+        method (str): Time stack creation method [identity]
+        output_path (str): A path to write the file to
+    '''
+
+    output_nodata = 2 ** 15 - 1
+    output_datatype = numpy.uint16
+
+    all_bands = _read_in_bands(image_paths)
+    if method is 'identity':
+        output_bands, mask = _mean_with_uniform_weight(
+            all_bands, output_nodata, output_datatype)
+    _write_out_bands(output_bands, mask, output_path, output_nodata)
+
+    return output_path
+
+
+def _read_in_bands(image_paths):
+    ''' Reads in a list of image paths and outputs a list of numpy arrays
+    representing the bands.
+
+    Input:
+        image_paths (list of str): A list of the input file paths
+
+    Output:
+        all_bands (list of list of arrays): A list of each band,
+            each band consisting of a list of images each entry being a
+            masked array of the image data
+    '''
+
+    all_images, nodata = _read_in_gdal_images(image_paths)
+    all_bands = _organise_images_to_bands(all_images, nodata)
+
+    return all_bands
+
+
+def _read_in_gdal_images(image_paths):
+    ''' Uses GDAL to read in the image as a numpy array and check them.
+
+    Input:
+        image_paths (list of str): A list of the input file paths
+
+    Output:
+        all_images (list of arrays of numbers): A list of each image,
+            each image consisting of a stacked array of the bands
+        nodata (number): The nodata value for this raster set
+    '''
+
+    # Find some constants
+    test_image_ds = gdal.Open(image_paths[0])
+    no_bands = test_image_ds.RasterCount
+    cols = test_image_ds.RasterXSize
+    rows = test_image_ds.RasterYSize
+    nodata = test_image_ds.GetRasterBand(1).GetNoDataValue()
+
+    # Read in each image as a numpy array
+    all_images = []
+    for image_path in image_paths:
+        image_ds = gdal.Open(image_path)
+
+        logging.info('Processing ' + image_path + '...')
+        assert image_ds.RasterCount == no_bands, \
+            '%s has a different number of bands' % image_path
+        assert image_ds.RasterXSize == cols and image_ds.RasterYSize == rows, \
+            '%s has a different size' % image_path
+        assert image_ds.GetRasterBand(1).GetNoDataValue() == nodata, \
+            '%s has a different nodata value' % image_path
+        alpha_band = image_ds.GetRasterBand(image_ds.RasterCount)
+        assert alpha_band.GetColorInterpretation() != gdal.GCI_AlphaBand, \
```

Comment:
Why are images with alpha bands not supported? We could use the alpha band in the weighting calculation